File: Latihan3/Latihan3_PHP/Pc.php
<?php

include "Processor.php";
include "Disk.php";
include "Ram.php";

class Pc
{

	// class attribute
	private $processor;
	private $disk;
	private $ram;
	private $totalPrice;

	// constructors
	public function Pc($processor, $disk, $ram)
	{
		$this->processor = $processor;
		$this->disk = $disk;
		$this->ram = $ram;
		$this->totalPrice = $processor->getPrice() + $disk->getPrice() + $ram->getPrice();
	}

	// getter and setter
	public function setProcessor($processor)
	{
		$this->processor = $processor;
	}

	public function getProcessor()
	{
		return $this->processor;
	}

	public function setDisk($disk)
	{
		$this->disk = $disk;
	}
	public function getDisk()
	{
		return $this->disk;
	}

	public function setRam($ram)
	{
		$this->ram = $ram;
	}

	public function getRam()
	{
		return $this->ram;
	}

	public function setTotalPrice($processorPrice, $diskPrice, $ramPrice)
	{
		$this->totalPrice = $processorPrice + $diskPrice + $ramPrice;
	}

	public function getTotalPrice()
	{
		return $this->totalPrice;
	}

	// show class method
	public function ShowClass()
	{
		echo "	PC SPECS" . "<br>";
		echo "PC Price	: " . $this->totalPrice . "<br>";
		echo "================================" . "<br>";
		echo "	PROCESSOR" . "<br>";
		echo "================================" . "<br>";
		echo "Processor Name	: " . $this->getProcessor()->getName() . "<br>";
		echo "Processor Price	: " . $this->getProcessor()->getPrice() . "<br>";
		echo "================================" . "<br>";
		echo "	DISK" . "<br>";
		echo "================================" . "<br>";
		echo "Disk Type	: " . $this->getDisk()->getType() . "<br>";
		echo "Disk Capacity	: " . $this->getDisk()->getCapacity() . "<br>";
		echo "Disk Price	: " . $this->getDisk()->getPrice() . "<br>";
		echo "================================" . "<br>";
		echo "	RAM" . "<br>";
		echo "================================" . "<br>";
		echo "RAM Capacity	: " . $this->getRam()->getCapacity() . "<br>";
		echo "RAM Price	: " . $this->getRam()->getPrice() . "<br>";
	}
}

File: Latihan3/Latihan3_Python/Pc.py
from dis import dis
from Processor import Processor
from Disk import Disk
from Ram import Ram


class Pc:
    # class attribute
    __processor = Processor()
    __disk = Disk()
    __ram = Ram()
    __totalPrice = 0

    # constructors
    def __init__(self, processor, disk, ram):
        self.__processor = processor
        self.__disk = disk
        self.__ram = ram
        self.__totalPrice = processor.getPrice() + disk.getPrice() + ram.getPrice()

    # getter and setter
    def setProcessor(self, processor):
        self.__processor = processor

    def getProcessor(self):
        return self.__processor

    def setDisk(self, disk):
        self.__disk = disk

    def getDisk(self):
        return self.__disk

    def setRam(self, ram):
        self.__ram = ram

    def getRam(self):
        return self.__ram

    def setTotalPrice(self, processorPrice, diskPrice, ramPrice):
        self.__totalPrice = processorPrice + diskPrice + ramPrice

    def getTotalPrice(self):
        return self.__totalPrice

    # shows class method
    def ShowClass(self):
        print("	PC SPECS")
        print("PC Price	: " + str(self.__totalPrice))
        print("==========================")
        print("	PROCESSOR")
        print("==========================")
        print("Processor Name	: " + self.__processor.getName())
        print("Processor Price	: " + str(self.__processor.getPrice()))
        print("==========================")
        print("	DISK")
        print("==========================")
        print("Disk Type	: " + self.__disk.getType())
        print("Disk Capacity: " + self.__disk.getCapacity())
        print("Disk Price	: " + str(self.__disk.getPrice()))
        print("==========================")
        print("	RAM")
        print("==========================")
        print("RAM Capacity	: " + self.__ram.getCapacity())
        print("RAM Price	: " + str(self.__ram.getPrice()))
